server/controllers/SolutionController.js

```diff
@@ -132,61 +124,72 @@ exports.getComments = async (req, res) => {
 };
 
 exports.create = async (req, res) => {
-  const { challengeId, title, description } = req.body;
-  const { filename: diagram } = req.file;
+  try {
+    const { challengeId, title, description } = req.body;
+    let diagram = null;
 
-  const userId = req.user.username;
+    if (req.file) {
+      diagram = req.file.filename; // Store just the filename, not the full path
+    }
 
-  console.log(`Creating Solution /w ${challengeId}m ${userId}`);
+    const userId = req.user.username;
 
-  const newSolution = await Solution.create({
-    challengeId,
-    userId: userId,
-    title,
-    description,
-    diagram,
-  });
+    console.log(`Creating Solution /w ${challengeId}, ${userId}`);
 
-  const challenge = await Challenge.findByPk(challengeId);
+    const newSolution = await Solution.create({
+      challengeId,
+      userId: userId,
+      title,
+      description,
+      diagram,
+    });
 
-  // Update user score
-  try {
-    const user = await User.findByPk(userId);
-    if (!user) {
-      return res.status(404).json({ message: "User not found" });
-    }
+    const challenge = await Challenge.findByPk(challengeId);
+
+    // Update user score
+    try {
+      const user = await User.findByPk(userId);
+      if (!user) {
+        return res.status(404).json({ message: "User not found" });
+      }
 
-    if (challenge.difficulty === "easy") {
-      user.score += 10;
-    } else if (challenge.difficulty === "medium") {
-      user.score += 20;
-    } else if (challenge.difficulty === "hard") {
-      user.score += 30;
-    } else {
-      console.error('Error finding challenge difficulty: ', challenge.difficulty);
+      if (challenge.difficulty === "easy") {
+        user.score += 10;
+      } else if (challenge.difficulty === "medium") {
+        user.score += 20;
+      } else if (challenge.difficulty === "hard") {
+        user.score += 30;
+      } else {
+        console.error('Error finding challenge difficulty: ', challenge.difficulty);
+      }
+
+      await user.save();
+    } catch (error) {
+      console.error('Error updating user score:', error);
+      return res.status(500).send('Server error');
     }
 
-    await user.save();
+    res.status(201).json(newSolution);
+
+    // Prepare AI feedback and submit it as a comment!
```

Comment:
The AI stuff is done after the solution is officially submitted since it takes a long time. Please put this back where it was :(